src/js/components/Calendar.js

```diff
@@ -285,24 +301,43 @@ export default class Calendar extends Component {
   }
 
   render () {
-    var classes = [CLASS_ROOT];
-    if (this.state.dropActive) {
-      classes.push(CLASS_ROOT + "--active");
-    }
-    if (this.props.className) {
-      classes.push(this.props.className);
-    }
+    const {
+      className,
+      id,
+      name,
+      value,
+      onChange, // eslint-disable-line no-unused-vars
+      ...props
+    } = this.props;
+
+    const classes = classnames(
+      CLASS_ROOT,
+      className,
+      {
+        [`${CLASS_ROOT}--active`]: this.state.dropActive
+      }
+    );
 
     return (
-      <div ref="component" className={classes.join(' ')}>
-        <input className={CLASS_ROOT + "__input"}
-          id={this.props.id} ref="calendarInput" name={this.props.name}
-          value={this.props.value}
-          onChange={this._onInputChange} />
-        <Button className={CLASS_ROOT + "__control"} icon={<CalendarIcon
-          a11yTitle='calendar-icon-title'
-          a11yTitleId='calendar-icon-title-id' />}
-          onClick={this._onOpen} />
+      <div ref="component" className={classes} {...props}>
```

Comment:
If you take a look at Box, when the Box is appCentered, we put the restProps on the outer div but the id and className props get put on the inner Component. I could see that being problematic as the caller adding a data-x property would probably expect it to be placed on the same DOM element as the id. Agreed that the on* event handlers should be on input. Perhaps the right rule of thumb is to transfer the props to wherever we put the id prop.